lib/rules/lint-table-groups.js

```diff
@@ -24,13 +24,24 @@ function hasDisallowedChildren(tableNode) {
       childrenFromValidBlocks,
       tableNode.children.filter(item => validBlocks.indexOf(item) === -1)
     )
-    .some(
-      child =>
-        !(
-          (child.type === 'ElementNode' && ALLOWED_TABLE_CHILDREN.indexOf(child.tag) > -1) ||
-          (child.type === 'TextNode' && !/\S/.test(child.chars))
-        )
-    );
+    .some(child => {
+      let tagNamePair;
+
+      switch (child.type) {
```

Comment:
This function is getting complicated so let's extract this switch statement into a new function `isAllowedTableChild(node)`. Then we can simply use it like: `.every(isAllowedTableChild)`. Note that I'm intentionally suggesting `isAllowedTableChild` instead of `isDisallowedTableChild` to get rid of the negative.

Reply:
updated